mergebot_backend/github_helper.py

```diff
@@ -155,14 +169,46 @@ def post_info(self, content, logger):
         """
         self.post_pr_comment('Info: {}'.format(content), logger)
 
+    def post_commit_status(self, state, url, description, context, logger):
+        """ Posts a commit status update to this PR.
```

Comment:
It is actually attached to a particular commit, and is using the "Commit Status API" from GitHub.

https://developer.github.com/v3/repos/statuses/

Statuses need to be attached to specific commits since they are what represent the HEAD of a particular pull request. GitHub doesn't have any other way to do this.